abstract_specification.md

```diff
@@ -197,96 +193,96 @@ For instance, a purely filesystem-based implementation may support only relative
 
-- type - a `primitive` type, expressed as an Arrow type (e.g., `int64`, `float32`, etc), indicating the type of data contained within the array
-- shape - the shape of the array, i.e., number and length of each dimension
+* type: a `primitive` type, expressed as an Arrow type (e.g. `int64`, `float32`, etc), indicating the type of data contained within the array;
+* shape: the shape of the array, i.e. number and length of each dimension.
 
-All dimensions must have a positive, non-zero length, and there must be 1 or more dimensions. Where explicitly referenced in the API, the dimensions are named `soma_dim_N`, where `N` is the dimension number (eg, `soma_dim_0`), and elements are named `soma_data`.
+All dimensions must have a positive, non-zero length, and there must be 1 or more dimensions. Where explicitly referenced in the API, the dimensions are named `soma_dim_N`, where `N` is the dimension number (e.g. `soma_dim_0`), and elements are named `soma_data`.
 
-The default "fill" value for `SOMADenseNDArray` is the zero or null value of the array type (e.g., Arrow.float32 defaults to 0.0).
+The default "fill" value for `SOMADenseNDArray` is the zero or null value of the array type (e.g. `Arrow.float32` defaults to 0.0).
 
-> ℹ️ **Note** - on TileDB this is an dense array with `N` `int64` dimensions of domain [0, maxInt64), and a single attribute.
+> ℹ️ **Note**: on TileDB this is an dense array with `N` `int64` dimensions of domain [0, maxInt64), and a single attribute.
 
 ### SOMASparseNDArray
 
 `SOMASparseNDArray` is a sparse, N-dimensional array of `primitive` type, with offset (zero-based) integer indexing on each dimension. The `SOMASparseNDArray` has a user-defined schema, which includes:
 
-- type - a `primitive` type, expressed as an Arrow type (e.g., `int64`, `float32`, etc), indicating the type of data contained within the array
-- shape - the shape of the array, i.e., number and length of each dimension
+* type: a `primitive` type, expressed as an Arrow type (e.g. `int64`, `float32`, etc), indicating the type of data contained within the array;
+* shape: the shape of the array, i.e. number and length of each dimension.
 
-All dimensions must have a positive, non-zero length, and there must be 1 or more dimensions. Implicitly stored elements (ie, those not explicitly stored in the array) are assumed to have a value of zero. Where explicitly referenced in the API, the dimensions are named `soma_dim_N`, where `N` is the dimension number (eg, `soma_dim_0`), and elements are named `soma_data`.
+All dimensions must have a positive (in particular, non-zero) length, and there must be 1 or more dimensions. Implicitly stored elements (i.e. those not explicitly stored in the array) are assumed to have a value of zero. Where explicitly referenced in the API, the dimensions are named `soma_dim_N`, where `N` is the dimension number (e.g. `soma_dim_0`), and elements are named `soma_data`.
```

Comment:
I find this edit gets in the way of the fact that it is "non-zero" AND "positive".  The "non-zero" is not an aside/parenthetical

Reply:
not sure I agree here; positive implies nonzero

Reply:
💯 , there are negative, zero, and positive numbers

Reply:
> positive implies nonzero

not according to the oracle, [wikipedia](https://en.wikipedia.org/wiki/Integer#:~:text=An%20integer%20is%20positive%20if,as%20neither%20negative%20nor%20positive)

no matter who is correct, it is ambiguous to many readers, and worth clarifying IMHO

Reply:
> not according to the oracle, [wikipedia](https://en.wikipedia.org/wiki/Integer#:~:text=An%20integer%20is%20positive%20if,as%20neither%20negative%20nor%20positive)

you might want to read that one back again because that is absolutely what it says.

(I like the parenthetical form and think it is plenty non-ambiguous.)

Reply:
@bkmartinjr I am saying
> there are negative, zero, and positive numbers
and the oracle is saying
> An integer is positive if it is greater than [zero](https://en.wikipedia.org/wiki/0), and negative if it is less than zero. Zero is defined as neither negative nor positive.

These are aligned.

I included _(in particular, non-zero)_ just to emphasize for people who aren't on the same page with you, me, and the oracle.

I'm happy to remove  _(in particular, non-zero)_, if you see it as redundant, if that's your ask.

Reply:
@thetorpedodog - well, the good news is I misread _your_ comment, not the Oracle. The bad news, is I misread a comment!